src/Infrastructure.VS/ThreadHandling.cs

```diff
@@ -42,6 +42,12 @@ public class ThreadHandling : IThreadHandling
 
         public T Run<T>(Func<Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.Run(asyncMethod);
 
+        public Task RunAsync<T>(Func<Task<T>> asyncMethod) => ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
+        {
+            await Task.Yield(); // get off the caller's callstack.
+            await asyncMethod();
+        }).Task;
```

Comment:
By default the method returns JoinableTask which is in a VS assembly, but this interface is in Core so I'm returning only the Task.

Reply:
I don't really understand what the `JoinableTask` is doing, but I'm worried that if we consume the inner `Task` directly we might mess up how it works. Could we do something like this instead, so we only wrapping the function?

```
        public async Task RunAsync(Func<Task> asyncMethod) => await ThreadHelper.JoinableTaskFactory.RunAsync(asyncMethod);
```